service/controller/resources/vaultcrt/resource.go

```diff
@@ -103,10 +105,26 @@ func (r *Resource) Name() string {
 }
 
 func (r *Resource) issueCertificate(customObject v1alpha1.CertConfig) (string, string, string, error) {
+	var pkiID string
+	{
+		switch customObject.Spec.Cert.ClusterComponent {
+		case string(certs.EtcdCert):
+			fallthrough
+		case string(certs.Etcd1Cert):
+			fallthrough
+		case string(certs.Etcd2Cert):
+			fallthrough
+		case string(certs.Etcd3Cert):
+			pkiID = fmt.Sprintf("%s-etcd", key.ClusterID(customObject))
+		default:
+			pkiID = key.ClusterID(customObject)
+		}
+	}
+
 	c := vaultcrt.CreateConfig{
 		AltNames:      key.AltNames(customObject),
 		CommonName:    key.CommonName(customObject),
-		ID:            key.ClusterID(customObject),
+		ID:            pkiID,
```

Comment:
How does this behave on cluster upgrades? E.g. a cluster is upgraded from a version without this separate CA to one with the separate CA

Reply:
good point. This is nothing I'm considering so far.
Current goal Is to make cluster creation to work in a CAPI-ready way and only after that I will think about upgradability

Reply:
Okay - we should not merge here then until this has been tested. We have really f-ed up in the past by accidentally merging incompatible changes in cert-operator.